presto-spi/src/main/java/com/facebook/presto/spi/block/ByteArrayBlockBuilder.java

```diff
@@ -221,6 +233,9 @@ public Block copyRegion(int positionOffset, int length)
     {
         checkValidRegion(getPositionCount(), positionOffset, length);
 
+        if (!hasNonNullValue) {
```

Comment:
If `newValueIsNull` below is all `true`s, you could also return all-nulls RLE

this applies to copyRegion methods in all block implementations and certainly adds some small overhead, not sure we would want that. @dain ?

Reply:
I don't think it is worth it.  My guess is the check would hit so rarely it wouldn't be worth the cost.